docs: setup README (#255)
srai-lab/srai: README.md

* **Full pipelines** - full pipelines for different embedding approaches, pre-configured from `srai` components
* **Image data download and processing** - downloading and processing image data (eg. OSM tiles, etc.)

## Installation

To install `srai` simply run:

```bash
pip install srai
```

This will install the `srai` package and dependencies required by most of the use cases. There are several optional dependencies that can be installed to enable additional functionality. These are listed in the [optional dependencies](#optional-dependencies) section.

### Optional dependencies

The following optional dependencies can be installed to enable additional functionality:

* `srai[all]` - all optional dependencies
* `srai[osm]` - dependencies required to download OpenStreetMap data
* `srai[voronoi]` - dependencies to use Voronoi-based regionization method
* `srai[gtfs]` - dependencies to process GTFS data
* `srai[plotting]` - dependencies to plot graphs and maps
* `srai[torch]` - dependencies to use torch-based embedders

## Usage

### Downloading OSM data

To download OSM data for a given area, using a set of tags use one of `OSMLoader` classes:

> * `OSMOnlineLoader` - downloads data from OpenStreetMap API - this is faster for smaller areas or tags counts

Reviewer: Maybe we should add info that it's a wrapper over `osmnx`?

Author: Good point ;)

> * `OSMPbfLoader` - loads data from automatically downloaded PBF file - this is faster for larger areas or tags counts

Reviewer: Here we can add info that we download those `*.osm.pbf` files from https://protomaps.com/

Author: Sure, will add it

Example with `OSMPbfLoader`:

```python
from srai.loaders import OSMOnlineLoader
from srai.utils import geocode_to_region_gdf

filter = {"leisure": "park"}
area = geocode_to_region_gdf("Wrocław, Poland")
loader = OSMOnlineLoader()

> parks_gdf = loader.load(area, filter)

Reviewer: We should probably change `filter` to some other name (`query`?) since filter is a keyword in python and syntax highlights it.

Author: Good point!

```python
from pathlib import Path

from srai.loaders import GTFSLoader

gtfs_path = Path("path/to/gtfs.zip")
loader = GTFSLoader()

features = loader.load(gtfs_path)
```

### Regionization

Regionization is a process of dividing a given area into smaller regions. This can be done in a variety of ways:

* `H3Regionizer` - regionization using [Uber's H3 library](https://github.com/uber/h3)
* `S2Regionizer` - regionization using [Google's S2 library](https://github.com/google/s2geometry)
* `VoronoiRegionizer` - regionization using Voronoi diagram
* `AdministativeBoundaryRegionizer` - regionization using administrative boundaries

Example:

```python
from srai.regionizers import H3Regionizer
from srai.utils import geocode_to_region_gdf

area = geocode_to_region_gdf("Wrocław, Poland")
regionizer = H3Regionizer(resolution=8)

regions = regionizer.transform(area)
```

### Embedding

Embedding is a process of mapping regions into a vector space. This can be done in a variety of ways:

* `Hex2VecEmbedder` - embedding using hex2vec[1] algorithm
* `GTFS2VecEmbedder` - embedding using GTFS2Vec[2] algorithm

> * `CountEmbedder` - embedding using count of features

Reviewer: This sentence seems odd for me. This
```suggestion
* `CountEmbedder` - embedding based on features counts

```

> * `ContextualCountEmbedder` - embedding using count of features in a given context (proposed in [3])

Reviewer: ```suggestion
* `ContextualCountEmbedder` - embedding based on features counts with neighbourhood context (proposed in [3])

```

* `Highway2VecEmbedder` - embedding using Highway2Vec[4] algorithm

All of those methods share the same API. All of them require results from `Loader` (load features), `Regionizer` (split area into regions) and `Joiner` (join features to regions) to work. An example using `CountEmbedder`:

```python
from srai.embedders import CountEmbedder
from srai.joiners import IntersectionJoiner
from srai.loaders import OSMPbfLoader
from srai.regionizers import H3Regionizer
from srai.utils import geocode_to_region_gdf

loader = OSMPbfLoader()
regionizer = H3Regionizer(resolution=9)
joiner = IntersectionJoiner()

area = geocode_to_region_gdf("Wrocław, Poland")

> features = loader.load(area)

Reviewer: Missing filter/query here

Author: Right

regions = regionizer.transform(area)
joint = joiner.transform(regions, features)

embedder = CountEmbedder()
embeddings = embedder.transform(regions, features, joint)
```

`CountEmbedder` is a simple method, which does not require fitting. Other methods, such as `Hex2VecEmbedder` or `GTFS2VecEmbedder` require fitting and can be used in a similar way to `scikit-learn` estimators:

```python
from srai.embedders import Hex2VecEmbedder
from srai.joiners import IntersectionJoiner
from srai.loaders import OSMPbfLoader
from srai.regionizers import H3Regionizer

loader = OSMPbfLoader()
regionizer = H3Regionizer(resolution=9)
joiner = IntersectionJoiner()

area = geocode_to_region_gdf("Wrocław, Poland")

> features = loader.load(area)

Reviewer: Missing tags/query here
